drivers/SmartThings/zigbee-window-treatment/src/screen-innovations/init.lua

```diff
@@ -0,0 +1,174 @@
+  local go_to_level = device.preferences.presetPosition or device:get_field(window_preset_defaults.PRESET_LEVEL_KEY) or window_preset_defaults.PRESET_LEVEL
+  -- send levels without inverting as: 0% closed (i.e., open) to 100% closed (Bug #16054)
+  device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, go_to_level))
+end
+
+-- this is device_added
+local function device_added(driver, device)
+  device:emit_event(capabilities.windowShade.supportedWindowShadeCommands({ "open", "close", "pause" }, {visibility = {displayed = false}}))
+  device.thread:call_with_delay(3, function(d)
+    update_device_info(device)
+  end)
+end
+
+-- this is current_position_attr_handler
+local function current_position_attr_handler(driver, device, value, zb_rx)
+  local level = value.value --Bug #16054
+  local event = nil
+
+  -- when the device is in action
+  if running_direction == motor_states.OPENING then
+    event = capabilities.windowShade.windowShade.opening()
+  end
+
+  if running_direction == motor_states.CLOSING then
+    event = capabilities.windowShade.windowShade.closing()
+  end
+
+  -- when the device is in idle
+  if running_direction == motor_states.IDLE then
+    if level == 0 then
+      event = capabilities.windowShade.windowShade.open() --Bug #16054
+    elseif level == 100 then
+      event = capabilities.windowShade.windowShade.closed() --Bug #16054
+    else
+      event = capabilities.windowShade.windowShade.partially_open()
+    end
+  end
+
+  -- update status
+  if event ~= nil then
+    device:emit_event(event)
+  end
+
+  -- update level
+  device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+end
+
+-- this is motor running_direction_attr_handler
+local function running_direction_attr_handler(driver, device, value, zb_rx)
+  local status = value.value
+  if status == 1 then
+    running_direction = motor_states.OPENING
+  elseif status == 2 then
+    running_direction = motor_states.CLOSING
+  else
+    running_direction = motor_states.IDLE
+  end
+end
+
+-- this is do_configure
+local function do_configure(self, device)
+  -- configure elements
+  device:configure()
+  device:send(device_management.build_bind_request(device, WindowCovering.ID, self.environment_info.hub_zigbee_eui))
+  device:send(WindowCovering.attributes.CurrentPositionLiftPercentage:configure_reporting(device, 1, 3600, 1))
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, self.environment_info.hub_zigbee_eui))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(device, 1, 3600, 1))
+  device:send(device_management.build_bind_request(device, Basic.ID, self.environment_info.hub_zigbee_eui))
+  device:send(Basic.attributes.PowerSource:configure_reporting(device, 1, 3600))
+
+  -- read elements
+  device.thread:call_with_delay(3, function(d)
+    device:send(Basic.attributes.ApplicationVersion:read(device))
```

Comment:
nothing handles the report of this attribute

Reply:
Removed the reporting of unhandled attribute.